tensorflow_datasets/core/custom_dataset/image_folder.py

```diff
@@ -0,0 +1,131 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Image FileFolder datasets."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import itertools
+import os
+
+from absl import logging
+import tensorflow.compat.v2 as tf
+from tensorflow_datasets import core
+
+_SUPPORTED_IMAGE_FORMAT = ('.jpg', '.jpeg', '.png')
+_AUTOTUNE = tf.data.experimental.AUTOTUNE
+
+
+class ImageLabelFolder(core.DatasetBuilder):
+
+  VERSION = core.Version(
+    "2.0.0", "New split API (https://tensorflow.org/datasets/splits)")
+
+  def __init__(self, data_dir):
+    self._data_dir = os.path.expanduser(data_dir)
+
+    # dict[split_name][label_name] = list(img_paths)
+    self._split_label_images, self._labels = self._get_split_label_images()
+    self._splits_info = core.SplitDict(self.name)
+
+    # Calculate splits for DatasetInfo
+    self._splits = {
+        split_name: self._get_split_size(split_name)
+        for split_name in self._split_label_images
+    }
+
+    self._split_dict = core.SplitDict(self.name)
+    for split_name, size in self._splits.items():
+      self._split_dict.add(
+        core.SplitInfo(name=split_name, shard_lengths=[size])) # is this correct?
+
+    super(ImageLabelFolder, self).__init__(
+        data_dir=data_dir, version=str(self.VERSION), config=None)
+
+    # _data_dir should not change to DATA_DIR/Version
+    self._data_dir = os.path.expanduser(data_dir)
+
+  def _get_split_label_images(self):
+    split_names = _list_folders(self.data_dir)
+    split_label_images = dict()
+    labels = set()
+    for split_name in split_names:
+      split_dir = os.path.join(self.data_dir, split_name)
+      split_label_images[split_name] = {
+          label_name: _list_imgs(os.path.join(split_dir, label_name))	
+          for label_name in _list_folders(split_dir)	
+      }
+      labels.update(split_label_images[split_name].keys())
+    return split_label_images, labels
+
+  def _get_split_size(self, split_name):
+    return sum(len(l) for l in self._split_label_images[split_name].values())
+
+  def _info(self):
+    ds_info = core.DatasetInfo(
+        builder=self,
+        description="Generic image classification dataset.",
+        # Generic features before the data is generated
+        features=core.features.FeaturesDict({
+            "image": core.features.Image(), # TODO: Image shape
+            "label": core.features.ClassLabel(names=list(self._labels)),
+        }),
+        supervised_keys=("image", "label"),
+    )
+    ds_info.update_splits_if_different(self._split_dict)
```

Comment:
Can you remove `self._split_dict` and add `self.info. update_splits_if_different()` in `__init__` ?

Reply:
Done